### ISSUE TITLE
codegenNativeCommands, ReadonlyArray Unsupported param type error

### ISSUE THREAD

**Issue by ysfzrn** [facebook/react-native#36817]
### Description

I am creating a fabric native component and adding native commands to this component. I haven't encountered any problems with simple data types or complex view props. However, when I define object types within an array as parameters for the addPolyLine command, I get the following error from codegen. Is this a bug or am I doing something wrong?
I couldn't find an example related to this in the documentation. (I'm using typeScript)
How can I define the codegenNativeCommands function with an array or object?

Sample Code
------
```js
export interface OrangeMobileMapViewProps extends ViewProps {
  theme?: WithDefault<string, 'dark'>;
  showsUserLocation?: boolean;
  polylineOptions?: Readonly<PolyLine>;
  overlay?: Readonly<Overlay>;
  onMapReady?: DirectEventHandler<null>;
  onRegionChangeComplete?: BubblingEventHandler<
    Readonly<{ latitude: Double; longitude: Double }>
  >;
}

export type OrangeMobileMapViewComponent =
  HostComponent<OrangeMobileMapViewProps>;

export interface OrangeMobileMapViewNativeCommands {
  animateToRegion: (
    viewRef: React.ElementRef<OrangeMobileMapViewComponent>,
    latitude: Float,
    longitude: Float,
    latitudeDelta: Float,
    longitudeDelta: Float,
    duration: Int32
  ) => void;
  addMarker: (
    viewRef: React.ElementRef<OrangeMobileMapViewComponent>,
    id: string,
    title: string,
    snippet: string,
    latitude: Float,
    longitude: Float,
    imageURL?: string
  ) => void;
  addPolyLine: (
    viewRef: React.ElementRef<OrangeMobileMapViewComponent>,
    id: string,
    route: ReadonlyArray<Readonly<{ latitude: Float; longitude: Float }>>,
    width: Int32,
    color: string,
    geodesic: boolean
  ) => void;
}

export const OrangeMobileMapViewCommands =
  codegenNativeCommands<OrangeMobileMapViewNativeCommands>({
    supportedCommands: ['animateToRegion', 'addMarker', 'addPolyLine'],
  });

export default codegenNativeComponent<OrangeMobileMapViewProps>(
  'OrangeMobileMapView'
) as OrangeMobileMapViewComponent;

```

### React Native Version

0.71.5

### Output of `npx react-native info`

```
Error: Unsupported param type for method "addPolyLine", param "route". Found ReadonlyArray
    at /Users/yusuf.zeren/Documents/workspace/@trendyol/orange-mobile-map/example/node_modules/react-native-codegen/lib/parsers/typescript/components/commands.js:88:15
    at Array.map (<anonymous>)
    at buildCommandSchema (/Users/yusuf.zeren/Documents/workspace/@trendyol/orange-mobile-map/example/node_modules/react-native-codegen/lib/parsers/typescript/components/commands.js:43:44)
    at /Users/yusuf.zeren/Documents/workspace/@trendyol/orange-mobile-map/example/node_modules/react-native-codegen/lib/parsers/typescript/components/commands.js:112:22
    at Array.map (<anonymous>)
    at getCommands (/Users/yusuf.zeren/Documents/workspace/@trendyol/orange-mobile-map/example/node_modules/react-native-codegen/lib/parsers/typescript/components/commands.js:112:6)
    at buildComponentSchema (/Users/yusuf.zeren/Documents/workspace/@trendyol/orange-mobile-map/example/node_modules/react-native-codegen/lib/parsers/typescript/components/index.js:264:20)
    at buildSchemaFromConfigType (/Users/yusuf.zeren/Documents/workspace/@trendyol/orange-mobile-map/example/node_modules/react-native-codegen/lib/parsers/utils.js:169:34)
    at buildSchema (/Users/yusuf.zeren/Documents/workspace/@trendyol/orange-mobile-map/example/node_modules/react-native-codegen/lib/parsers/typescript/index.js:65:10)
    at parseFile (/Users/yusuf.zeren/Documents/workspace/@trendyol/orange-mobile-map/example/node_modules/react-native-codegen/lib/parsers/utils.js:135:10)

```

### Steps to reproduce

- Try to define the parameters of any function created with codegenNativeCommands as an array or object.

### Snack, code example, screenshot, or link to a repository

-

**Comment by ysfzrn** [comment]
<img width="744" alt="Screenshot 2023-04-05 at 23 18 58" src="https://user-images.githubusercontent.com/15312525/230200690-57574d0f-9719-4aa1-9c42-f757e99ecea5.png">

Apparently, there is no Array in the supported types of the codegenNativeCommands array. I hope, in the future, it will support the Array type.

**Comment by cipolleschi** [comment]
Hi @ysfzrn! We know that array support for events and commands is not existent right now. We are tracking the task and working on that internally.
We hope to be able to release something soon!

**Comment by ysfzrn** [comment]
Hi @cipolleschi thanks for answer. The new architecture is going great, I'm sure you'll add this soon as well. Good luck!

**Comment by ravindraguptacapgemini** [comment]
@cipolleschi do we have the support for array now for events and commands?

**Comment by ravindraguptacapgemini** [comment]
@cipolleschi is it specifically added in 0.72.4, I could not find it working in 0.72.3, tried to use ReadOnlyArray and Array both.

**Comment by cipolleschi** [comment]
You are right. The change landed on main on May, the 4th. So it was not cherry picked into 0.72 as it was not a fix but a new feature. They will be available starting from 0.73
